honeybee_radiance/properties/face.py

```diff
@@ -0,0 +1,181 @@
+
+        Args:
+            data: A dictionary representation of FaceRadianceProperties.
+            host: A Face object that hosts these properties.
+        """
+        assert data['type'] == 'FaceRadianceProperties', \
+            'Expected FaceRadianceProperties. Got {}.'.format(data['type'])
+
+        try:  # ensure the putil module is imported, which imports all primitive modules
+            putil
+        except NameError:
+            import honeybee_radiance.putil as putil
+
+        new_prop = cls(host)
+        if 'modifier' in data and data['modifier'] is not None:
+            new_prop.modifier = putil.dict_to_modifier(data['modifier'])
+        if 'modifier_blk' in data and data['modifier_blk'] is not None:
+            new_prop.modifier_blk = putil.dict_to_modifier(data['modifier_blk'])
+        return new_prop
+
+    def apply_properties_from_dict(self, abridged_data, modifiers):
+        """Apply properties from a FaceRadiancePropertiesAbridged dictionary.
+
+        Args:
+            abridged_data: A FaceRadiancePropertiesAbridged dictionary (typically
+                coming from a Model).
+            modifiers: A dictionary of modifiers with modifier names as keys,
+                which will be used to re-assign modifiers.
+        """
+        if 'modifier' in abridged_data and abridged_data['modifier'] is not None:
+            self.modifier = modifiers[abridged_data['modifier']]
+        if 'modifier_blk' in abridged_data and abridged_data['modifier_blk'] is not None:
+            self.modifier_blk = modifiers[abridged_data['modifier_blk']]
+
+    def to_dict(self, abridged=False):
+        """Return radiance properties as a dictionary.
+
+        Args:
+            abridged: Boolean to note whether the full dictionary describing the
+                object should be returned (False) or just an abridged version (True).
+                Default: False.
+        """
+        base = {'radiance': {}}
+        base['radiance']['type'] = 'FaceRadianceProperties' if not \
+            abridged else 'FaceRadiancePropertiesAbridged'
+        if self._modifier is not None:
+            base['radiance']['modifier'] = \
+                self._modifier.name if abridged else self._modifier.to_dict()
+        if self._modifier_blk is not None:
+            base['radiance']['modifier_blk'] = \
+                self._modifier_blk.name if abridged else self._modifier_blk.to_dict()
+        return base
+
+    def duplicate(self, new_host=None):
+        """Get a copy of this object.
+
+        new_host: A new Face object that hosts these properties.
+            If None, the properties will be duplicated with the same host.
+        """
+        _host = new_host or self._host
+        return FaceRadianceProperties(_host, self._modifier, self._modifier_blk)
+
+    def ToString(self):
+        return self.__repr__()
+
+    def __repr__(self):
+        return 'Face Radiance Properties:\n host: {}'.format(self.host.name)
```

Comment:
It looks to me that we can save a lot of very similar code by using a base class for aperture, door, shade and face. We may just need to add a new method to modifier-set to take the type name. Is there another reason that you are not using a baseclass?

Reply:
I wasn't sure whether we would need these classes to be more different once we implemented states. Now that I understand that a state is like a whole other copy of the object (e. Aperture, Shade) with its own modifiers, geometry, and assigned objects changed, I think we can have a base class here.

Reply:
Done!